pyface/ui/qt4/font.py

```diff
@@ -0,0 +1,199 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+"""
+Font conversion utilities
+
+This module provides facilities for converting between pyface Font objects
+and Qt QFont objects, trying to keep as much similarity as possible between
+them.
+"""
+
+from pyface.qt.QtGui import QFont
+
+
+qt_family_to_generic_family = {
+    QFont.AnyStyle: 'default',
+    QFont.System: 'default',
+    QFont.Decorative: 'fantasy',
+    QFont.Serif: 'serif',
+    QFont.Cursive: 'cursive',
+    QFont.SansSerif: 'sans-serif',
+    QFont.Monospace: 'monospace',
+    QFont.TypeWriter: 'typewriter',
+}
+generic_family_to_qt_family = {
+    'default': QFont.System,
+    'fantasy': QFont.Decorative,
+    'decorative': QFont.Decorative,
+    'serif': QFont.Serif,
+    'roman': QFont.Serif,
+    'cursive': QFont.Cursive,
+    'script': QFont.Cursive,
+    'sans-serif': QFont.SansSerif,
+    'swiss': QFont.SansSerif,
+    'monospace': QFont.Monospace,
+    'modern': QFont.Monospace,
+    'typewriter': QFont.TypeWriter,
+}
+
+weight_to_qt_weight = {
+    100: 0,
+    200: QFont.ExtraLight,
+    300: QFont.Light,
+    400: QFont.Normal,
+    500: QFont.Medium,
+    600: QFont.DemiBold,
+    700: QFont.Bold,
+    800: QFont.ExtraBold,
+    900: QFont.Black,
+    1000: 99,
+}
+qt_weight_to_weight = {
+    0: 'thin',
+    QFont.ExtraLight: 'extra-light',
+    QFont.Light: 'light',
+    QFont.Normal: 'normal',
+    QFont.Medium: 'medium',
```

Comment:
Isn't this `"medium"` not allowed in the `weight` trait? 
In fact, could we map all of these to `"100"`, `"200"` and so on? Then `QFont.Medium` will be mapped to `"500"`, which is allowed.

Reply:
Getting back to this after a year, since this is just the descriptions with no attempt at font-name parsing, `'medium'` should be allowed both here and in the weight trait as there is no conflict with other uses of `medium` at this level.